docs/source/format/how_manager.rst

```diff
@@ -0,0 +1,204 @@
+mechanisms.  For example, the Go and C# implementations of ADBC can both
+export drivers to consumers who expect the C API definitions.  As long as the
+definitions in adbc.h are implemented somehow, then the application is
+generally none the wiser when it comes to what's actually underneath.
+
+How does an application call these functions, though?  Here, there are several
+options.
+
+Again, the simplest case is as follows: if (1) the application links directly
+to the driver, and (2) the driver exposes the ADBC functions *under the same
+name* as in adbc.h, then the application can just ``#include <adbc.h>`` and
+call ``AdbcStatementExecuteQuery(...)`` directly.  Here, the application and
+driver have a relationship no different than any other C library.
+
+.. figure:: DriverDirectLink.mmd.svg
+
+   In the simplest case, an application directly links to the driver and calls
+   ADBC functions.  When the application calls ``StatementExecuteQuery``, that
+   is directly provided by the driver that it links against.
+
+Unfortunately, this doesn't work as well in other scenarios.  For example, if
+an application wishes to use multiple ADBC drivers, this no longer works: both
+drivers define the same functions (the ones in adbc.h), and when the
+application links both of them, the linker has no way of telling which
+driver's function is meant when the application calls an ADBC function.  On
+top of that, this violates the `One Definition Rule`_.
+
+In this case, the driver can provide driver-specific aliases that applications
+can use, say ``PostgresqlStatementExecuteQuery`` or
+``FlightSqlStatementExecuteQuery``.  Then, the application can link both
+drivers, ignore the ``Adbc…`` functions (and ignore the technical violation of
+the One Definition Rule there), and use the aliases instead.
+
+.. figure:: DriverAlias.mmd.svg
+
+   To get around the One Definition Rule, we can provide aliases of the ADBC
+   APIs instead.
+
+This is rather inconvenient for the application, though.  Additionally, this
+sort of defeats the point of using ADBC, since now the application has a
+separate API for each driver, even if they're technically all clones of the
+same API.  And this doesn't solve the problem for applications that want to
+load drivers dynamically.  For example, a Python script would want to load the
+driver at runtime.  In that case, it would need to know which functions from
+the driver correspond to which functions in the ADBC API definitions, without
+having to hardcode this knowledge.
+
+ADBC anticipated this, and defined :cpp:struct:`AdbcDriver`.  This is just a
+table of function pointers with one entry per ADBC function.  That way, an
+application can dynamically load a driver and call an entrypoint function that
+returns this table of function pointers.  (It does have to hardcode or guess
+the name of the entrypoint; the ADBC spec lists a set of names it can try,
+based on the name of the driver library itself.)
```

Comment:
We should probably explain why we didn't just *only* do this, what benefit do we get from allowing simple case over just requiring this every time

Reply:
Well, because you can link to the driver and use it. Do we have to spell it out more than we already did?

Reply:
I added this comment before I saw the additional stuff below it. I think you've covered it pretty well